### PR TITLE
#170947444 User view his/her  trip requests

### PR DESCRIPTION
**Pull request by blaise82**
### What does this PR do?

- create GET request route
- create get request method in the trip controller
- write tests

### How should this be manually tested?

- clone this repo using `git clone https://github.com/andela/octopus-bn-backend`
- run `cd octopus-bn-backend` to move to the directory
- run `npm install` to install all required dependencies
- run `npm run dev` to start the development server
- open postman GET localhost:3000/api/v1/trips/request

### Implementation plan
[Integrate the requests table to the backend](https://docs.google.com/document/d/1k32rRF6f1LAUQPGX5IvHRH0mPTb51qzFUDkWe_ZZEA8/edit)

### Screenshots if any
Screen Shot 2020-02-24 at 11 15 10

[finishes #170947444]

### CODE REVIEW

File: locales/en.js [hunk 1]
```diff
@@ -14,32 +14,44 @@
...
+	"validFromCountry": "Please provide a valid from country: eg [KN, Rw, USA, UG, ...]",
+	"validFromCity": "Please provide a valid from city: eg [Kigali, kampala, ....]",
+	"validToCountry": "Please provide a valid to country: eg [USA, RW, UG, ...]",
+	"validToCity": "Please provide a valid to CITY: eg [Kigali, kampala, ....]",
+	"validAccommodation": "Please provide a valid accommodation value: [yes, no]",
+	"validReturnOn": "Please provide valid date for return, date before or on",
+	"onlyReturn": "Only provide a return date on a return trip request",
+	"ValidReason": "Please provide a reason (more than 20 char)",
+	"retrievedSuccessfully": "Requests retrieved successfully"
+}
```
Comment (hakoemmy) [inline locales/en.js]:
```suggestion
}

```

File: locales/en.json [hunk 2]
```diff
@@ -14,32 +14,44 @@
...
+	"validFromCountry": "Please provide a valid from country: eg [KN, Rw, USA, UG, ...]",
+	"validFromCity": "Please provide a valid from city: eg [Kigali, kampala, ....]",
+	"validToCountry": "Please provide a valid to country: eg [USA, RW, UG, ...]",
+	"validToCity": "Please provide a valid to CITY: eg [Kigali, kampala, ....]",
+	"validAccommodation": "Please provide a valid accommodation value: [yes, no]",
+	"validReturnOn": "Please provide valid date for return, date before or on",
+	"onlyReturn": "Only provide a return date on a return trip request",
+	"ValidReason": "Please provide a reason (more than 20 char)",
+	"retrievedSuccessfully": "Requests retrieved successfully"
+}
```
Comment (hakoemmy) [inline locales/en.json]:
```suggestion
}

```

File: locales/fr.js [hunk 3]
```diff
@@ -1,44 +1,55 @@
...
+    "validFromCountry": "Veuillez fournir un pays valide: par exemple [KN, Rw, USA, UG, ...]",
+    "validFromCity": "Veuillez fournir un valide de la ville: par exemple [Kigali, kampala, ....]",
+    "validToCountry": "Veuillez fournir un pays valide: par exemple [USA, RW, UG, ...]",
+    "validToCity": "Veuillez fournir un valide à CITY: par exemple [Kigali, kampala, ....]",
+    "validAccommodation": "Veuillez fournir une valeur d'hébergement valide: [oui, non]",
+    "validReturnOn": "Veuillez fournir une date de retour valide, une date avant ou le",
+    "onlyReturn": "Fournissez uniquement une date de retour sur une demande de retour",
+    "ValidReason": "Veuillez fournir une raison (plus de 20 caractères)",
+    "retrievedSuccessfully": "Demandes de déplacement récupérées avec succès"
+}
```
Comment (hakoemmy) [inline locales/fr.js]:
```suggestion
}

```

File: locales/fr.json [hunk 4]
```diff
@@ -1,44 +1,55 @@
...
+    "validFromCountry": "Veuillez fournir un pays valide: par exemple [KN, Rw, USA, UG, ...]",
+    "validFromCity": "Veuillez fournir un valide de la ville: par exemple [Kigali, kampala, ....]",
+    "validToCountry": "Veuillez fournir un pays valide: par exemple [USA, RW, UG, ...]",
+    "validToCity": "Veuillez fournir un valide à CITY: par exemple [Kigali, kampala, ....]",
+    "validAccommodation": "Veuillez fournir une valeur d'hébergement valide: [oui, non]",
+    "validReturnOn": "Veuillez fournir une date de retour valide, une date avant ou le",
+    "onlyReturn": "Fournissez uniquement une date de retour sur une demande de retour",
+    "ValidReason": "Veuillez fournir une raison (plus de 20 caractères)",
+    "retrievedSuccessfully": "Demandes de déplacement récupérées avec succès"
+}
```
Comment (hakoemmy) [inline locales/fr.json]:
```suggestion
}

```

File: src/controllers/tripsController.js [hunk 5]
```diff
@@ -131,6 +125,21 @@ class tripsController {
...
+  static async getTrips(req, res) {
+    const { userID } = req.user;
+    try {
+      const trips = await travelRequests.findAll({
+        where: { userID },
+        attributes: {
+          exclude: ['id', 'userID', 'passportNumber', 'gender', 'createdAt', 'updatedAt']
+        },
+      });
+      return successResponse(res, 200, req.i18n.__('retrievedSuccessfully'), trips);
```
Comment (hakoemmy) [inline src/controllers/tripsController.js]:
```suggestion
      return successResponse(res, 200, setLanguage(req.user.preferedLang).__('retrievedSuccessfully'), trips);
```
@blaise82  I think that you'll have to keep track of translation basing on the user's preference when my PR is probably merged.

File: src/controllers/tripsController.js [hunk 6]
```diff
@@ -131,6 +125,21 @@ class tripsController {
...
+    } catch (error) {
+      return errorResponse(res, error.status || 500, error.message);
```
Comment (hakoemmy) [inline src/controllers/tripsController.js:144]:
```suggestion
      return errorResponse(res, error.status || 500, setLanguage(req.user.preferedLang).__('internalServerError'));
```
Would you please consider returning a specific message to a user other than server default errors?
The locale for ```internalServerError``` key has been already set.

File: src/middlewares/tripRequestDateValidator.js [hunk 7]
```diff
@@ -5,12 +5,9 @@ import JoiCountryExtension from 'joi-country-extension';
 const Joi = BaseJoi.extend(Extension, JoiCountryExtension);
 const dateValidator = (req, res, next) => {
   const dataSchema = Joi.object().keys({
-    departureDate: Joi.date().format('YYYY-MM-DD').error(() => ({
-      message: 'Date format must be YYYY-MM-DD'
-    })),
 
     returnDate: Joi.date().format('YYYY-MM-DD').error(() => ({
-      message: 'Date format must be YYYY-MM-DD'
+      message: req.i18n.__('validReturnDate')
```
Comment (hakoemmy) [inline src/middlewares/tripRequestDateValidator.js]:
```suggestion
      message: setLanguage(req.user.preferedLang).__('validReturnDate')
```

File: src/middlewares/tripRequestValidator.js [hunk 8]
```diff
@@ -8,52 +8,48 @@ const Joi = BaseJoi.extend(Extension, JoiCountryExtension);
...
-        message: 'Only provide a return date on a return trip request'
+        message: req.i18n.__('onlyReturn')
       })),
     }),
 
     reason: Joi.string().min(20).required().error(() => ({
-      message: 'Please provide a reason (more than 20 char)'
+      message: req.i18n.__('ValidReason')
     })),
   });
```
Comment (hakoemmy) [inline src/middlewares/tripRequestValidator.js]: When my PR is merged. make use of ```setLanguage(req.user.preferedLang).__('YourKey')```

File: src/routes/api/trips.routes.js [hunk 9]
```diff
@@ -136,5 +136,5 @@ router.post('/request', [checkUser, dateValidator, tripRequestValidator], tripsC
  *         description: Travel request successfully created
  */
 router.post('/multi-city', validateMultiCity, checkUser, tripsController.createMultiCityTrip);
-
+router.get('/request', checkUser, tripsController.getTrips);
```
Comment (hakoemmy) [inline src/routes/api/trips.routes.js]: @blaise82  I've created middleware called ```isProfileUpdated.js```. You must accept it While you'll be rebasing.

Comment (hakoemmy) [inline src/routes/api/trips.routes.js]:
```suggestion
router.get('/', checkUser, tripsController.getTrips);
```
Just a suggestion, I know in the creation of a one-way trip request you are using ```/request``` but if you change  GET request to ```/api/v1/trips``` that would look nice.

**Review by hakoemmy — CHANGES_REQUESTED** [review]

File: src/controllers/tripsController.js (hunk 5 above)
Comment (blaise82) [reply src/controllers/tripsController.js]: that haven't been merged
so I don't think it applies now

File: src/controllers/tripsController.js (hunk 6 above)
Comment (blaise82) [reply src/controllers/tripsController.js:144]: That sounds better

File: src/controllers/tripsController.js (hunk 5 above)
Comment (hakoemmy) [reply src/controllers/tripsController.js]: @blaise82 For sure. You'll address these changes when this [PR](https://github.com/andela/octopus-bn-backend/pull/28) is merged.

**Review by hakoemmy — APPROVED** [review]

**Review by RusimbiPatrick — APPROVED** [review]

**Review by AbdoulNuru — APPROVED** [review]

**Review by kcharles52 — APPROVED** [review]

**Pull request merged**